src/tfvc/tfvc-extension.ts

```diff
@@ -52,6 +53,35 @@ export class TfvcExtension  {
             "Checkin");
     }
 
+    /**
+     * This command runs a delete command on the selected file.  It gets a Uri object from vscode.
+     */
+    public async Delete(uri?: Uri): Promise<void> {
+        this.displayErrors(
+            async () => {
+                if (uri) {
+                    const basename: string = path.basename(uri.fsPath);
+                    try {
+                        const message: string = `Are you sure you want to delete '${basename}'?`;
+                        if (await UIHelper.PromptForConfirmation(message, Strings.DeleteFile)) {
+                            Telemetry.SendEvent(this._repo.IsExe ? TfvcTelemetryEvents.DeleteExe : TfvcTelemetryEvents.DeleteClc);
+                            await this._repo.Delete([uri.fsPath]);
+                        }
+                    } catch (err) {
+                        //Provide a better error message if the file to be deleted isn't in the workspace (e.g., it's a new file)
+                        if (err.tfvcErrorCode && err.tfvcErrorCode === TfvcErrorCodes.FileNotInWorkspace) {

```

Comment:
Can we check ahead of time and just do a normal delete for them? I.e. so the user can ALWAYS call our delete.